k8-operator/controllers/infisicalsecret_controller.go

```diff
@@ -107,6 +107,18 @@ func (r *InfisicalSecretReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		api.API_HOST_URL = infisicalSecretCR.Spec.HostAPI
 	}
 
+	if infisicalSecretCR.Spec.TLS.CaRef.SecretName != "" {
+		api.API_CA_CERTIFICATE, err = r.GetInfisicalCaCertificateFromKubeSecret(ctx, infisicalSecretCR)
```

Comment:
How will this work if you have multiple infisical instances (staging, prod, etc) and you use the single cluster? It looks like this will be the same api.API_CA_CERTIFICATE for all Infisical CRDs?

Reply:
if you have multiple infisical instances, you would need multiple `InfisicalSecret` CRDs with different configs for the `HOST_URL` and `CA`.

the reconcile loop runs independently between different InfisicalSecrets so there should be no issue here

Reply:
>  runs independently between different InfisicalSecrets

Hmm are you sure? This would then mean we can get rid of this pattern https://github.com/Infisical/infisical/blob/main/k8-operator/controllers/infisicalsecret_controller.go#L49